Dockerfile

```diff
@@ -1,11 +1,13 @@
 FROM node:14-alpine
 
 WORKDIR /app
-COPY . .
 
 RUN apk add --no-cache --virtual .build-deps alpine-sdk python3 git openjdk8-jre cmake
+COPY package.json package-lock.json ./
+RUN npm config set unsafe-perm true && npm install
+COPY . .
 RUN echo "GIT_TAG=$(git tag --points-at HEAD)" >> .env
-RUN npm config set unsafe-perm true && npm install && npm run build && npm prune --production
+RUN npm run build && npm prune --production
```

Comment:
I don't understand the effective changes implemented here. It looks like some Dockerfile directives were moved around and/or split into multiple directives. Can you clarify why this is needed?